### DISCUSSION TITLE
Fuzzing plugins in oss-fuzz with --engine afl

### DISCUSSION THREAD

**Discussion by predator89090** [zeek/zeek#3882] [Help] [answered]
Hello, I would like to know if anyone has tried fuzz testing plugins via oss-fuzz with the afl engine? What value of map density is normal?

I’ll give an example of testing the built-in mysql plugin
![image](https://github.com/user-attachments/assets/47ff5ba7-3aa4-4ec4-ab26-5a4d3702fbb7)

These indicators are considered acceptable when fuzzing plugins? Based on what I read in the afl documentation, these indicators are considered low, or did I understand something wrong?

**Comment by timwoj** [comment, marked as answer]
I don't believe we have. We're configured to use [libFuzzer and hongfuzz](https://github.com/google/oss-fuzz/blob/f0c125b017aea47d84992343f7e4b4c8198c3c62/projects/zeek/project.yaml#L14-L16), but never investigated whether AFL gave us anything extra. If I remember correctly from looking at AFL in another project long ago, it was more complicated to setup, but that's not a reason we didn't configure it for Zeek.
